src/txmempool.h

```diff
@@ -60,6 +61,8 @@ class CTxMemPoolEntry
     unsigned int GetHeight() const { return nHeight; }
     bool WasClearAtEntry() const { return hadNoDependencies; }
     size_t DynamicMemoryUsage() const { return nUsageSize; }
+
+    uint32_t GetBranchId() const { return nBranchId; }
```

Comment:
This assumes the transaction has a branchid.
Didn't we change it to have a group version id instead?
And wasn't the branch_id in the transaction supposed to for formatting and potentially different from the consensus branchid?

Reply:
This is not the version group ID (which is stored in the transaction if present, and can trivially be fetched). This is the branch ID corresponding to the consensus rules under which the transaction has been validated, which we cache here so that when the mineable branch ID changes, transactions in the mempool which won't be valid under it can be identified and dropped.

Reply:
Note that a branch ID of zero corresponds to no upgrades being active, ie. the Sprout consensus rules are in effect, so for pre-Overwinter transactions `GetBranchId()` will return zero.

Reply:
It isn't obvious which branch ID it should be returning, though. Please add documentation and/or rename to ``GetValidatedBranchId()``.

Reply:
There's only one branch ID now. But ACK, will rename.

Reply:
I'll also update the documentation of `CTxMemPoolEntry::nBranchId` to clarify it is a cached value we have elsewhere checked to be valid.